app/src/crash/crash-app.tsx

```diff
@@ -0,0 +1,176 @@
+import * as React from 'react'
+import { ipcRenderer, shell, remote } from 'electron'
+import { ICrashDetails, ErrorType } from './shared'
+import { TitleBar } from '../ui/window/title-bar'
+import { WindowState, getWindowState, windowStateChannelName } from '../lib/window-state'
+import { Octicon, OcticonSymbol } from '../ui/octicons'
+import { Button } from '../ui/lib/button'
+import { getVersion } from '../ui/lib/app-proxy'
+import { getOS } from '../lib/get-os'
+
+interface ICrashAppProps {
+  readonly startTime: number
+}
+
+interface ICrashAppState {
+  readonly type?: ErrorType
+  readonly error?: Error
+  readonly windowState: WindowState
+}
+
+// Note that we're reusing the welcome illustrations here, any changes to these
+// will have to be reflected in the welcome flow as well.
+const BottomImageUri = `file:///${__dirname}/static/welcome-illustration-left-bottom.svg`
+
+function prepareErrorMessage(error: Error) {
+
+  let message
```

Comment:
Yeah, this is actually new from TS 2.1 https://blogs.msdn.microsoft.com/typescript/2016/11/08/typescript-2-1-rc-better-inference-async-functions-and-more/#user-content-smarter-inference